src/core/services/file.ts

```diff
@@ -834,9 +834,9 @@ export class CoreFileProvider {
             return this.copyOrMoveExternalFile(from, to, copy);
         }
 
-        const moveCopyFn: MoveCopyFunction = copy ?
-            (isDir ? File.copyDir.bind(File.instance) : File.copyFile.bind(File.instance)) :
-            (isDir ? File.moveDir.bind(File.instance) : File.moveFile.bind(File.instance));
+        const moveCopyFn = (path: string, name: string, newPath: string, newName: string) => copy ?
+            (isDir ? File.copyDir(path, name, newPath, newName) : File.copyFile(path, name, newPath, newName)) :
+            (isDir ? File.moveDir(path, name, newPath, newName) : File.moveFile(path, name, newPath, newName));
```

Comment:
```suggestion
        const moveCopyFn = (...args: [string, string, string, string]) => copy ?
            (isDir ? File.copyDir(...args) : File.copyFile(...args)) :
            (isDir ? File.moveDir(...args) : File.moveFile(...args));
```

This is another way of doing it, although in both cases there's something funky going on with the response type, maybe we should take another look.